include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,33 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @brief Port / Endpoint DT Helpers
+ *
+ */
+
+/* Drivers should not use this except they know exactly that port has a pid */
+#define _DT_INST_PORT_BY_ID(n, pid)                                                                \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(n, ports)), (DT_CHILD(DT_INST_CHILD(n, ports), port_##pid)), (DT_INST_CHILD(n, port_##pid)))
+
+/* Drivers rarely use this as they usually deals with only endpoint */
+#define DT_INST_PORT_BY_ID(n, pid)                                                                 \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(n, pid)), (_DT_INST_PORT_BY_ID(n, pid)), (DT_INST_CHILD(n, port)))
+
+/* Drivers should not use this except they know exactly that the endpoint has a pid */
+#define _DT_INST_ENDPOINT_BY_ID(n, pid, id) DT_CHILD(DT_INST_PORT_BY_ID(n, pid), endpoint_##id)
+
+#define DT_INST_ENDPOINT_BY_ID(n, pid, id)                                                         \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_ENDPOINT_BY_ID(n, pid, id)), (_DT_INST_ENDPOINT_BY_ID(n, pid, id)), (DT_CHILD(DT_INST_PORT_BY_ID(n, pid), endpoint)))
+
+/* Drivers should not use this except they know exactly that the remote device has no ports node */
+#define DT_REMOTE_DEVICE_NO_PORTS(ep)                                                              \
+	DT_GPARENT(DT_NODELABEL(DT_STRING_TOKEN(ep, remote_endpoint_label)))
+
+#define DEVICE_DT_GET_REMOTE_DEVICE(ep)                                                            \
+	COND_CODE_1(DT_NODE_EXISTS(DT_CHILD(DT_PARENT(DT_REMOTE_DEVICE_NO_PORTS(ep)), ports)), \
+(DEVICE_DT_GET(DT_PARENT(DT_REMOTE_DEVICE_NO_PORTS(ep)))), (DEVICE_DT_GET(DT_REMOTE_DEVICE_NO_PORTS(ep))))
```

Comment:
A macro with `DEVICE_DT_` prefix would likely go to [`<zephyr/device.h>`](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/device.h). This means it could be interesting to reduce the macro that would ultimately go to `<zephyr/devicetree.h>` down to the DT manipulation:

```suggestion
#define DT_REMOTE_DEVICE(ep)                                                            \
	COND_CODE_1(DT_NODE_EXISTS(DT_CHILD(DT_PARENT(DT_REMOTE_DEVICE_NO_PORTS(ep)), ports)), \
(DT_PARENT(DT_REMOTE_DEVICE_NO_PORTS(ep))), (DT_REMOTE_DEVICE_NO_PORTS(ep)))
```

Reply:
Then, `DEVICE_DT_GET_REMOTE_DEVICE(ep)` using it (fated to land in `<zephyr/device.h>`) can be defined like this:

```c
#define DEVICE_DT_GET_REMOTE_DEVICE(ep) DEVICE_DT_GET(DT_REMOTE_DEVICE(ep))
```

Reply:
EDIT: I changed my mind: no added value in the suggestion I proposed.